requirements.txt

```diff
@@ -1,31 +1,31 @@
 -e .[http2,brotli]
```

Comment:
So... we'd need the deps that this introduces to be pinned, right?

*(To resolve this I'd start by looking up how we did this with `httpcore`)* 🤷‍♂️

Reply:
Hmm, it follows the same as `httpcore`: https://github.com/encode/httpcore/pull/184/files

Reply:
Okay, right. Seems like an oversight there too, then. (I think, right?)

Reply:
Hmm, is it? :thinking:
It feels weird to pin the extra requirements :thinking: I mean, the tests **needs** to pass with the latest version allowed by the version range on `setup.py`, right? :thinking:

Reply:
Ahhhhhh.... right. Which helps tease out an interesting point.

(I'm glad we've taken this super slowly, because I don't think we'd have noticed it otherwise)

I'd kinda agree with your assessment here...

* We probably want our *tooling* to be pinned, because it's an environment we can strictly control, and it'll give more reproducible build results.
* We possibly don't want our *package dependencies* to be pinned, because they won't be for our users, and we probably may as well just *always* be installing vs. whatever's the latest.

If that *is* what we opt for with the pinning, then adding caching https://github.com/encode/httpx/pull/1680 probably *isn't* something we'd want to do. Because it'd mean we'll *sometimes* get the latest package dependancies, and sometimes not get them, depending on that state of the cache. Or it might mean that we *are* okay with caching, but we only cache the tooling deps, and we always make sure to update to the very latest dependencies.

Either way though, I think we should resolve what we do or don't want for the pinning, and get this PR resolved, because this is still a valid PR regardless of the outcome of #1680.

Reply:
> If that is what we opt for with the pinning, then adding caching #1680 probably isn't something we'd want to do. Because it'd mean we'll sometimes get the latest package dependancies, and sometimes not get them, depending on that state of the cache. Or it might mean that we are okay with caching, but we only cache the tooling deps, and we always make sure to update to the very latest dependencies.

Well, we can also invalidate the cache within a few days (say `7 days`). But not caching the requirements on `setup.py` also works, ofc. 

Anyway, I guess we should continue this discussion on #1680